### DISCUSSION TITLE
How does Antimicrox intercept controller key press / stick events?

### DISCUSSION THREAD

**Discussion by aykaramba** [AntiMicroX/antimicrox#982] [Q&A]
I am using Antimicrox on Linux + Xorg and really enjoy it.  Great project.  

I don't understand how this software intercepts controller key / stick input and remaps them to key combinations.   How is this done?  I don't know very much C/C++ but it looks like there is a daemon that listens to the key presses and then returns whatever they are set to be mapped to.

What is the basic concept of how this is done on an Xorg system conceptually?   How are the controller keypresses intercepted?

I would really like to understand this key concept and start learning more about how this is done so I can start playing around a bit more.

Thanks!

**Comment by pktiuk** [comment]
Hi @aykaramba    

In general flow looks like this:

Library SDL2 reads inputs from gamepads -> Antimicrox processes them -> generated events are executed using one of the implemented event handlers https://github.com/AntiMicroX/antimicrox/tree/master/src/eventhandlers (xtest, uinput or winsend).
